src/aria2.cpp

```diff
@@ -84,24 +84,24 @@ Aria2::Aria2():
   }
   mp_aria = Subprocess::run(callCmd);
   mp_curl = curl_easy_init();
-  char errbuf[CURL_ERROR_SIZE];
+
   curl_easy_setopt(mp_curl, CURLOPT_URL, "http://localhost/rpc");
   curl_easy_setopt(mp_curl, CURLOPT_PORT, m_port);
   curl_easy_setopt(mp_curl, CURLOPT_POST, 1L);
-  curl_easy_setopt(mp_curl, CURLOPT_ERRORBUFFER, errbuf);
+  curl_easy_setopt(mp_curl, CURLOPT_ERRORBUFFER, m_curlErrorBuffer.get());
 
   int watchdog = 50;
   while(--watchdog) {
     sleep(10);
-    errbuf[0] = 0;
+    m_curlErrorBuffer[0] = 0;
     auto res = curl_easy_perform(mp_curl);
     if (res == CURLE_OK) {
       break;
     } else if (watchdog == 1) {
-      std::cerr <<" curl_easy_perform() failed." << std::endl;
-      fprintf(stderr, "\nlibcurl: (%d) ", res);
-      if (errbuf[0] != 0) {
-        std::cerr << errbuf << std::endl;
+      std::cerr <<"ERROR : curl_easy_perform() failed." << std::endl;
+      std::cerr << "libcurl: (" << res << ") " << std::endl;
+      if (m_curlErrorBuffer[0] != 0) {
+        std::cerr << m_curlErrorBuffer.get() << std::endl;
       } else {
         std::cerr << curl_easy_strerror(res) << std::endl;
       }
```

Comment:
I would deduplicate this piece of code with an identical piece of code in `Aria2::doRequest()`

Reply:
Done with a macro.